clients/fides-js/src/lib/cookie.ts

```diff
@@ -316,8 +316,10 @@ export const buildTcfEntitiesFromCookie = (
               Boolean(cookieConsent[item.id]),
               ConsentMechanism.OPT_IN
             )
-          : // if experience contains a tcf entity not defined by tcfEntities, we override experience current pref with the default pref
-            item.default_preference;
+          : // If experience contains a tcf entity not defined by tcfEntities, this means either:
+            // A) Most commonly, user has opted out, and opt-outs are not tracked by TC string. It's safe to assume this case.
+            // B) There is a new tcf entity that requires consent. In this case we would just resurface the banner
+            ConsentMechanism.OPT_OUT;
```

Comment:
I think we only want this for the fides string override case. in other words, I think this change would break this flow:
* No fides string override
* User opts out of all purposes
* A new legint purpose is added
* On load, the new legint purpose would be opted out, even though in this case, because we have the cookie key consent, we actually know the user has _not_ opted out.

I think this could be fixed with...

```js
const defaultPreference = cookie.fides_string ? ConsentMechanism.OPT_OUT : item.default_preference;
const preference = Object.hasOwn(cookieConsent, item.id) 
                 ? transformConsentToFidesUserPreference(
                      Boolean(cookieConsent[item.id]),
                      ConsentMechanism.OPT_IN
                    )
                 : defaultPreference;
```

Reply:
Ah this makes more sense, thanks @allisonking !